include/micm/configure/solver_config.hpp

```diff
@@ -361,51 +366,66 @@ namespace micm
       const std::string THIRD_BODY = "THIRD_BODY";
 
       ValidateSchema(object, { NAME, TYPE }, { TRACER_TYPE, ABS_TOLERANCE, DIFFUSION_COEFF, MOL_WEIGHT });
-
-      std::string name = object[NAME].get<std::string>();
+      
+      std::string name = object[NAME].as<std::string>();
       Species species{ name };
 
       // Load remaining keys as properties
-      for (auto& [key, value] : object.items())
+      for(auto it = object.begin(); it != object.end(); ++it)
       {
+        auto key = it->first.as<std::string>();
+        auto value = it->second;
+
         if (key != NAME && key != TYPE)
-        {
-          if (value.is_string())
+        {       
+          std::string stringValue = value.as<std::string>();
+          if (!stringValue.empty() && isInt(stringValue)) {
+            species.SetProperty<int>(key, value.as<int>());
+          }
+          else if (!stringValue.empty() && isFloat(stringValue)) {
+            species.SetProperty<double>(key, value.as<double>());
+          }
+          else if (!stringValue.empty() && isBool(stringValue))
+          {
+            species.SetProperty<bool>(key, value.as<bool>());
+          }
+          else
           {
-            if (key == TRACER_TYPE && value == THIRD_BODY)
+            if (key == TRACER_TYPE && stringValue == THIRD_BODY)
             {
               species.SetThirdBody();
             }
             else
             {
-              species.SetProperty<std::string>(key, value);
+              species.SetProperty<std::string>(key, stringValue);
             }
           }
-          else if (value.is_number_integer())
-          {
-            species.SetProperty<int>(key, value);
-          }
-          else if (value.is_number_float())
-          {
-            species.SetProperty<double>(key, value);
-          }
-          else if (value.is_boolean())
-          {
-            species.SetProperty<bool>(key, value);
-          }
-          else
-          {
-            throw std::system_error{ make_error_code(MicmConfigErrc::InvalidType), key };
-          }
-        }
+        } 
```

Comment:
I am fine with this change. Maybe I'd suggest
```c++
if (key != NAME && key != TYPE)
{       
  std::string stringValue = value.as<std::string>();
  if (!stringValue.empty())
  {
    if (isInt(stringValue)){...}
    else if (isFloat(stringValue)){...}
    else if (isBool(stringValue)) {...}
    else {throw error}
  }
  else if (key == TRACER_TYPE && stringValue == THIRD_BODY) {...}
  else 
  {
    species.SetProperty<std::string>(key, stringValue);
  }
}
```

Reply:
Now I see this again, I'm wondering why `stringValue.empty()` is checked only for `int`, `float`, `bool` type. Maybe it could be 
```C++
if (key != NAME && key != TYPE)
{       
  std::string stringValue = value.as<std::string>();
  if (stringValue.empty()) {throw error}
  
  if (isInt(stringValue)){...}
  else if (isFloat(stringValue)){...}
  else if (isBool(stringValue)) {...}
  else if (key == TRACER_TYPE && stringValue == THIRD_BODY) {...}
  else 
  {
    species.SetProperty<std::string>(key, stringValue);
  }
}
```